src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -524,18 +524,29 @@ export class StudyViewPageStore {
                             return lcValue;
                         });
                     }
+
                     statusCallback(LoadingPhase.CREATING_SESSION);
-                    // create groups using data
-                    const groups = _.map(lcValueToSampleIdentifiers, (sampleIdentifiers, lcValue)=>{
-                        const value = lcValueToValue[lcValue];
-                        return {
-                            name: value,
-                            description: "",
-                            studies: getStudiesAttr(sampleIdentifiers),
-                            origin: this.studyIds,
-                            color: lcValueToColor[lcValue].color,
-                        };
-                    });
+
+                    // create groups using data - up to MAX_GROUPS_IN_SESSION - ordered by size
+                    const groups:SessionGroupData[] = [];
+                    const sortedAttrVals = _.sortBy(clinicalAttributeValues, attrVal=>-attrVal.count);
+                    for (const attrVal of sortedAttrVals) {
+                        if (groups.length >= MAX_GROUPS_IN_SESSION) {
```

Comment:
may be move this break statement to end of loop

Reply:
imo it's good here. in principle moving it to the end could lead to overshooting it, if anything about the previous code changes